app/src/components/data-grid/autocomplete/AsyncAutocompleteDataGridEditCell.tsx

```diff
@@ -0,0 +1,204 @@
+import { mdiMagnify } from '@mdi/js';
+import Icon from '@mdi/react';
+import Autocomplete, { createFilterOptions } from '@mui/material/Autocomplete';
+import Box from '@mui/material/Box';
+import CircularProgress from '@mui/material/CircularProgress';
+import TextField from '@mui/material/TextField';
+import { GridRenderCellParams, GridValidRowModel, useGridApiContext } from '@mui/x-data-grid';
+import { IAutocompleteDataGridOption } from 'components/data-grid/autocomplete/AutocompleteDataGrid.interface';
+import { DebouncedFunc } from 'lodash-es';
+import { useEffect, useState } from 'react';
+
+export interface IAsyncAutocompleteDataGridEditCell<
+  DataGridType extends GridValidRowModel,
+  ValueType extends string | number
+> {
+  /**
+   * Data grid props for the cell.
+   *
+   * @type {GridRenderCellParams<DataGridType>}
+   * @memberof IAsyncAutocompleteDataGridEditCell
+   */
+  dataGridProps: GridRenderCellParams<DataGridType>;
+  /**
+   * Function that returns a single option. Used to translate an existing value to its matching option.
+   *
+   * @memberof IAsyncAutocompleteDataGridEditCell
+   */
+  getCurrentOption: (value: ValueType) => Promise<IAutocompleteDataGridOption<ValueType> | null>;
+  /**
+   * Search function that returns an array of options to choose from.
+   *
+   * @memberof IAsyncAutocompleteDataGridEditCell
+   */
+  getOptions: DebouncedFunc<
+    (
+      searchTerm: string,
+      onSearchResults: (searchResults: IAutocompleteDataGridOption<ValueType>[]) => void
+    ) => Promise<void>
+  >;
+}
+
+/**
+ * Data grid single value asynchronous autocomplete component for edit.
+ *
+ * @template DataGridType
+ * @template ValueType
+ * @param {IAsyncAutocompleteDataGridEditCell<DataGridType, ValueType>} props
+ * @return {*}
+ */
+const AsyncAutocompleteDataGridEditCell = <DataGridType extends GridValidRowModel, ValueType extends string | number>(
+  props: IAsyncAutocompleteDataGridEditCell<DataGridType, ValueType>
+) => {
+  const { dataGridProps, getCurrentOption, getOptions } = props;
+
+  const apiRef = useGridApiContext();
+
+  // The current data grid value
+  const dataGridValue = dataGridProps.value;
+  // The input field value
+  const [inputValue, setInputValue] = useState('');
+  // The currently selected option
+  const [currentOption, setCurrentOption] = useState<IAutocompleteDataGridOption<ValueType> | null>(null);
+  // The array of options to choose from
+  const [options, setOptions] = useState<IAutocompleteDataGridOption<ValueType>[]>([]);
+  // Is control loading (search in progress)
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (!dataGridValue) {
+      // No current value
+      return;
+    }
+
+    if (dataGridValue === currentOption?.value) {
+      // Existing value matches tracked value
+      return;
+    }
+
+    const fetchCurrentOption = async () => {
```

Comment:
Not sure what the purpose of this function is; It's instantiated and immediately called, and it's return value isn't used.

Reply:
I think this is just a pattern for making async calls inside of a useEffect